updater/reports/ReportRepoUsage.py

```diff
@@ -0,0 +1,53 @@
+from .ReportDaily import *
+
+# Calculate percentage of active repositories over the last four weeks
+# in organizations that have received pushes to at least one protected
+# branch and that have at least one topic assigned to them.
+class ReportRepoUsage(ReportDaily):
+	def name(self):
+		return "repository-usage"
+
+	def updateDailyData(self):
+		newHeader, newData = self.parseData(self.executeQuery(self.query()))
+		self.header = newHeader if newHeader else self.header
+		self.data.extend(newData)
+		self.truncateData(self.timeRangeTotal())
+		self.sortDataByDate()
+
+	def query(self):
+		oneDayAgo = self.yesterday()
+		fourWeeksAgo = self.daysAgo(28)
+		query = '''
+			SELECT
+				"''' + str(oneDayAgo) + '''" AS date,
+				ROUND(100*COUNT(DISTINCT(protect.repository_id))/COUNT(DISTINCT(active.repository_id)))
+					AS "protected branches [%]",
+				ROUND(100*COUNT(DISTINCT(topics.repository_id))/COUNT(DISTINCT(active.repository_id)))
+					AS "topics [%]"
+			FROM
+				(
+				SELECT
+					repositories.id AS repository_id,
+					ref
```

Comment:
Could you write `repositories.ref` (if that’s the right one)? I had trouble seeing where `ref` comes from.